src/formio/utils.js

```diff
@@ -45,4 +48,39 @@ const setErrorAttributes = (elements, hasErrors, hasMessages, messageContainerId
   });
 };
 
-export {applyPrefix, escapeHtml, setErrorAttributes};
+const linkToSoftRequiredDisplay = (elements, component) => {
+  // if soft required validation is not enabled, there's nothing to do
+  if (!component.component.openForms?.softRequired) return;
+
+  const isEmpty = component.isEmpty();
+
+  const softRequiredIds = [];
+  FormioUtils.eachComponent(component.root.components, component => {
+    if (component.type === 'softRequiredErrors') {
+      const id = `${component.id}-content`;
+      softRequiredIds.push(id);
+    }
+  });
+
+  // Update the attribute 'aria-describedby' based on whether the component is empty
+  elements.forEach(element => {
```

Comment:
yeah wasn't sure if this warranted a refactor or not since your helper also touches aria-invalid and mine doesn't so there is going to be some duplication, but I'll take a look.